src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -20,6 +21,16 @@ namespace System
     //       to also change the test class.
     internal static class ConsolePal
     {
+        private static readonly object s_cursorGate = new object();
```

Comment:
Some things in `ConsolePal.Unix.cs` lock on `Console.Out`. Perhaps it makes sense to use that instead of `s_cursorGate`.

Reply:
Yeah, I think it'd be better to share in the same lock.  We could revisit that if it ever proved to be problematic, but I'd rather err on that side than have multiple locks and risk deadlock if we ever accidentally invert the lock hierarchy order.